WebDriverAgentLib/Categories/XCElementSnapshot+FBXPath.m

```diff
@@ -0,0 +1,264 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "XCElementSnapshot+FBXPath.h"
+
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wpadded"
+#endif
+
+#import <libxml/tree.h>
+#import <libxml/parser.h>
+#import <libxml/HTMLparser.h>
+#import <libxml/xpath.h>
+#import <libxml/xpathInternals.h>
+#import <libxml/encoding.h>
+#import <libxml/xmlwriter.h>
+
+#ifdef __clang__
+#pragma clang diagnostic pop
+#endif
+
+#import "FBLogger.h"
+#import "FBXPathCreator.h"
+#import "XCAXClient_iOS.h"
+#import "XCTestDriver.h"
+#import "XCTestPrivateSymbols.h"
+#import "XCUIElement.h"
+#import "XCUIElement+FBWebDriverAttributes.h"
+
+const static char *_UTF8Encoding = "UTF-8";
+
+static NSString *const kXMLIndexPathKey = @"private_indexPath";
+static NSString *const topNodeIndexPath = @"top";
+NSString *const XCElementSnapshotInvalidXPathException = @"XCElementSnapshotInvalidXPathException";
+NSString *const XCElementSnapshotXPathQueryEvaluationException = @"XCElementSnapshotXPathQueryEvaluationException";
+
+@implementation XCElementSnapshot (FBXPathHelpers)
```

Comment:
Methods in this category are way to big. You should create separate class and split them in to smaller ones. If you also make then unaware of XCTest stuff then you can create a lot of unit tests around them and few integration tests which we already have. Otherwise if something breaks it will be scary to touch does.